core/src/main/java/com/linecorp/armeria/internal/common/AbstractSplitHttpMessage.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal.common;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+
+import org.reactivestreams.Subscriber;
+
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpMessage;
+import com.linecorp.armeria.common.SplitHttpMessage;
+import com.linecorp.armeria.common.stream.StreamMessage;
+import com.linecorp.armeria.common.stream.SubscriptionOption;
+import com.linecorp.armeria.internal.common.stream.NoopSubscription;
+
+import io.netty.util.concurrent.EventExecutor;
+
+abstract class AbstractSplitHttpMessage implements SplitHttpMessage, StreamMessage<HttpData> {
+
+    private static final AtomicIntegerFieldUpdater<AbstractSplitHttpMessage> subscribedUpdater =
+            AtomicIntegerFieldUpdater.newUpdater(AbstractSplitHttpMessage.class, "subscribed");
+
+    private volatile int subscribed;
+
+    private final HttpMessage upstream;
+    private final EventExecutor upstreamExecutor;
+    private final SplitHttpMessageSubscriber bodySubscriber;
+
+    AbstractSplitHttpMessage(HttpMessage upstream, EventExecutor upstreamExecutor,
+                                       SplitHttpMessageSubscriber bodySubscriber) {
```

Comment:
nit: indentation

Reply:
Thank you :) I fixed it in https://github.com/line/armeria/pull/3953/commits/f580416d73887a08b739a48c4f9f2a5a10daf021